src/core/sys/darwin/mach/thread_act.d

```diff
@@ -135,3 +140,103 @@ version (i386)
     kern_return_t thread_resume(thread_act_t);
     kern_return_t thread_get_state(thread_act_t, thread_state_flavor_t, thread_state_t*, mach_msg_type_number_t*);
 }
+else version (AppleARM)
+{
+    alias thread_act_t = mach_port_t;
+    alias thread_state_t = void;
+    alias thread_state_flavor_t = int;
+    alias mach_msg_type_number_t = natural_t;
+
+    enum
+    {
+        ARM_THREAD_STATE = 1,
+        ARM_UNIFIED_THREAD_STATE = ARM_THREAD_STATE,
+        ARM_VFP_STATE = 2,
+        ARM_EXCEPTION_STATE = 3,
+        ARM_DEBUG_STATE = 4, /* pre-armv8 */
+        THREAD_STATE_NONE = 5,
+        ARM_THREAD_STATE64 = 6,
+        ARM_EXCEPTION_STATE64 = 7,
+        // ARM_THREAD_STATE_LAST = 8, /* legacy */
+        ARM_THREAD_STATE32 = 9
+    }
+
+    enum
+    {
+        ARM_DEBUG_STATE32 = 14,
+        ARM_DEBUG_STATE64 = 15,
+        ARM_NEON_STATE = 16,
+        ARM_NEON_STATE64 = 17,
+        ARM_CPMU_STATE64 = 18
+    }
+
+    enum
+    {
+        ARM_AMX_STATE = 24,
+        ARM_AMX_STATE_V1 = 25
+    }
+
+    struct arm_thread_state_t
```

Comment:
This is identical to `arm_thread_state32_t`, why not provide a simple alias?
```suggestion
    alias arm_thread_state_t = arm_thread_state32_t;
```

Reply:
Fixed.